cli/src/commands/squash.rs

```diff
@@ -356,14 +361,24 @@ pub(crate) fn cmd_squash(
                     )?;
                     // It's weird that commit.description() contains "JJ: " lines, but works.
                     commit_builder.set_description(combined);
+                    already_edited = true;
                     let temp_commit = commit_builder.write_hidden()?;
                     let intro = "Enter a description for the combined commit.";
                     let template = description_template(ui, &tx, intro, &temp_commit)?;
                     edit_description(&text_editor, &template)?
                 }
             }
         };
-        commit_builder.set_description(new_description);
```

Comment:
Perhaps, we'll need to update `Exact|UseDestination` cases to do `add_trailers()` unconditionally if editor is forcibly enabled. `description.is_empty()` is special cased only when the final description is guaranteed to be empty.